software/SolarAISVN/SolarAISVN.ino

```diff
@@ -161,14 +164,18 @@ void measureVoltages() {
   //    32,      33,       34,       35,   14,   26,   27,     12,   13
   // solar, battery, currentA, currentB, load, wind, dump, solar2, LiPo  
   voltage[0] = int((4096 - voltage[0]) * 7.52 - 1000);  // pin32 solar    voltage divider 10k : 1.2 k Ohm 1:1
+  if(voltage[0] < 0) voltage[0] = 0;
   voltage[1] = int((4096 - voltage[1]) * 7.52 - 1000);  // pin33 battery  voltage divider 10k : 1.2 k Ohm 1:1
   //voltage[2] = int((voltage[2]) * 0.804 + 129);         // pin34 voltage solar minus green LED
   //voltage[3] = int((voltage[3]) * 0.804 + 129);         // pin35 voltage solar minus green LED minus 0.1 Ohm serial
   voltage[2] = int((voltage[3] - voltage[2]) * 6.75);   // voltage difference pin35 - pin34 x 8.4 is corrent (x0.804)
   voltage[4] = int((4096 - voltage[4]) * 7.52 - 1000);  // pin14 load     voltage divider 10k : 1.2 k Ohm 1:1
+  if(voltage[4] < 0) voltage[4] = 0;  
   voltage[5] = int((4096 - voltage[5]) * 7.52 - 1000);  // pin26 wind     voltage divider 10k : 1.2 k Ohm 1:1
-  voltage[6] = int((voltage[6]) * 1.608 + 258);         // pin27 dump     not connected yet
+  if(voltage[5] < 0) voltage[5] = 0;  
+  voltage[6] = int((voltage[6]) * 0.804 + 129);         // pin27 temp     not connected yet
```

Comment:
Included some specific calculation to have no negative voltages measured

Reply:
Professional solution.